### ISSUE TITLE
adding live stream crashes on links longer than 50 characters

### ISSUE THREAD

**Issue by slash851** [kaltura/kmc-ng#671]
When Creating new HLS livestream and pasting URL that is longer that 50 characters whole KMC crashes.
Tried with this URL:
https://hls11.webcamera.pl/zakopane_cam_cd5b2d/zakopane_cam_cd5b2d.stream/playlist.m3u8

**Steps to reproduce:**
1. Click  CREATE button
2. Choose to Create Live Stream Entry
3. Provide Name of new entry
4. Paste m3u8 url under Live Mobile HLS stream URL

**Expected behavior** 
- will allow creating a new entry

**Actual behavior** 
- after pasting the url whole page is froze and user is not able to do anything, page require to be refreshed

**Environment:**
OS: Win 10
Browser: Chrome Version 66.0.3359.181 (Official Build) (64-bit)

**Comment by slash851** [comment]
Just tested same on newest Firefox.
**Console log:**

content.js:4 [Deprecation] chrome.loadTimes() is deprecated, instead use standardized API: nextHopProtocol in Navigation Timing 2. https://www.chromestatus.com/features/5637885046816768.
(anonymous) @ content.js:4
content.js:5 [Deprecation] chrome.loadTimes() is deprecated, instead use standardized API: nextHopProtocol in Navigation Timing 2. https://www.chromestatus.com/features/5637885046816768.
(anonymous) @ content.js:5
66 Refused to load the font '<URL>' because it violates the following Content Security Policy directive: "font-src assets-cdn.github.com".

**Comment by kaltura-hooks** [comment]
Hi @slash851,

Thank for you reporting an issue and helping improve Kaltura!

To get the fastest response time, and help the maintainers review and test your reported issues or suggestions, please ensure that your issue includes the following (please comment with more info if you have not included all this info in your original issue):
 - Is the issue you're experiencing consistent and across platforms? or does it only happens on certain conditions?
    please provide as much details as possible.
 - Which Kaltura deployment you're using: Kaltura SaaS, or self-hosted?
	If self hosted, are you using the RPM or deb install?
 - Packages installed.
	When using RPM, paste the output for:
```
	# rpm -qa \"kaltura*\"
```	
	For deb based systems:
```	
	# dpkg -l \"kaltura-*\"
```
 - If running a self hosted ENV - provide the MySQL server version used
 - If running a self hosted ENV - is this a single all in 1 server or a cluster?
 - If running a self hosted ENV, while making the problematic request, run:
```
	# tail -f /opt/kaltura/log/*.log /opt/kaltura/log/batch/*.log | grep -A 1 -B 1 --color \"ERR:\|PHP\|trace\|CRIT\|\[error\]\"
```
   and paste the output.
 - When relevant, provide any screenshots or screen recordings showing the issue you're experiencing.

For general troubleshooting see:
https://github.com/kaltura/platform-install-packages/blob/Jupiter-10.13.0/doc/kaltura-packages-faq.md#troubleshooting-help

If you only have a general question rather than a bug report, please close this issue and post at:
http://forum.kaltura.org

Thank you in advance,

**Comment by esakal** [comment]
Hi @slash851 ,
Following the scenario you provided we managed to identify the problem. We are using a url validator to validate the provided url using [a regular expression we found on the web](https://github.com/kaltura/kaltura-ng/blob/master/kaltura-ui/src/validators/validators.ts#L4). Although it had a lot of stars it appears to be too complicated and once you provide a long url the javascript engine of the browser stop working and hang the browser session (hence you must force quit it). We will provide a fix with our coming version by using a more performant expression.

Since you provided us with console logs I assume you are running your own version of the kmcng? If so you will be able to transpile the code with the fix we tend to push later today. 

I will update you once the fix will be available.

Thanks for your input.
Eran.
